modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -353,13 +362,32 @@ protected void beforeCreateDocument(DocumentModel doc) {
         snapDoc = session.saveDocument(snapDoc);
         snapDoc = session.move(snapDoc.getRef(), session.getParentDocumentRef(tmp.getRef()), null);
 
-        // cleanup tmp dir
+        // we need to submit the move, then wait for its completion in case of async implementation
+        commitOrRollbackAndRestartTransaction();
+        try {
+            int timeout = 10;
+            if (!Framework.getService(BulkService.class).await(Duration.ofMinutes(timeout))) {
+                // Interrupt the import
+                throw new NuxeoException("NXDistribution import timed out after " + timeout
```

Comment:
No way to pinpoint the imported package causing the timeout in the exception ?

Reply:
Yes, in the upstream method. This is why I moved the log there:
https://github.com/nuxeo/nuxeo-explorer/pull/163/files#diff-9230fd9226b418f30ef485fbcd0604b5d2628abd135fc3d08ab7913d859cdbe8R319

Reply:
The `properties` hold several infos that are given at submission and it contains enough to identify the distrib (alias etc...)